python/ql/test/experimental/dataflow/tainttracking/defaultAdditionalTaintStep/test_syntax.py

```diff
@@ -23,6 +23,33 @@ def test_with():
     with ctx as tainted:
         ensure_tainted(tainted) # $ tainted
 
+class Context_taint:
+    def __enter__(self):
+        return TAINTED_STRING
+
+    def __exit__(self, exc_type, exc, tb):
+        pass
+
+def test_with_taint():
+    ctx = Context_taint()
+    with ctx as tainted:
+        ensure_tainted(tainted) # $ MISSING: tainted
+
+
+class Context_arg:
+    def __init__(self, arg):
+        self.arg = arg
+
+    def __enter__(self):
+        return self.arg
+
+    def __exit__(self, exc_type, exc, tb):
+        pass
+
+def test_with_arg():
+    ctx = Context_arg(TAINTED_STRING)
+    with ctx as tainted:
+        ensure_tainted(tainted) # $ tainted
```

Comment:
Can you explain why this one ends up with taint, when the `Context_taint()` one doesn't? (it's not clear to me)

I would have thought that it would only be `ctx.arg` that was tainted, and not `ctx` itself. But if `with ctx as x` can make `x` tainted from the implementation of `__enter__`, then it should also work for `Context_taint` :thinking: 

If you're also in doubt of how this works, maybe we can add a `ensure_not_tainted` like so: 


```suggestion
    ctx = Context_arg(TAINTED_STRING)
    ensure_not_tainted(ctx)
    with ctx as tainted:
        ensure_tainted(tainted) # $ tainted
```

Reply:
I believe the line
```python
ctx = Context_arg(TAINTED_STRING)
```
taints `ctx`.

I will run it as a path query to be absolutely sure..

Reply:
It does. The taint is passed to the constructor argument and then to (the post-update of) `self` then ending up at `Context_arg(TAINTED_STRING)`:

![Screenshot 2021-11-08 at 13 35 03](https://user-images.githubusercontent.com/647347/140743439-5451bb38-bfd1-4c37-bc4b-48ce4eccbbc2.png)

Reply:
that's absolutely correct. I am very surprised about this though, and absolutely didn't know this is how our taint modeling worked with attributes :open_mouth: (I'll be making a PR with some tests to highlight this).

Since this behavior is not introduced/changed in this PR, I don't want to make any more fuss about it here :blush: